doc/source/skips/4-transition-to-v2.rst

```diff
@@ -0,0 +1,254 @@
+scope and ambition of the 2.0 transition could grow if this SKIP is accepted.
+This SKIP instead proposes a mechanism for managing the transition without
+breaking users' code. A meta-issue tracking the proposed changes can be found
+on GitHub, scikit-image/scikit-image#5439 [7]_. Some examples are briefly
+included below for illustrative purposes:
+
+- Stop rescaling input arrays when the dtype must be coerced to float.
+- Stop swapping coordinate axis order in different contexts, such as drawing or
+  warping.
+- Allow automatic return of non-NumPy types, so long as they are coercible to
+  NumPy with ``numpy.asarray``.
+- Harmonizing similar parameters in different functions to have the same name;
+  for example, we currently have ``random_seed``, ``random_state``, ``seed``,
+  or ``sample_seed`` in different functions, all to mean the same thing.
+- Changing ``measure.regionprops`` to return a dictionary instead of a list.
+- Combine functions that have the same purpose, such as ``watershed``,
+  ``slic``, or ``felzenschwalb``, into a common namespace. This would make it
+  easier for new users to find out which functions they should try out for a
+  specific task. It would also help the community grow around common APIs,
+  where now scikit-image APIs are essentially unique for each function.
+
+To make this transition with a minimum amount of user disruption, this SKIP
+proposes releasing a new library, skimage2, that would replace the existing
+library, *but only if users explicitly opt-in*. Additionally, by releasing a
+new library, users could depend *both* on scikit-image (1.0) and on skimage2,
+allowing users to migrate their code progressively.
+
+Related Work
+------------
+
+`pandas` released 1.0.0 in January 2020, including many backwards-incompatible
+API changes [3]_. `scipy` released version 1.0 in 2017, but, given its stage of
+maturity and position at the base of the scientific Python ecosystem, opted not
+to make major breaking changes [4]_. However, SciPy has adopted a policy of
+adding upper-bounds on dependencies [5]_, acknowledging that the ecosystem as a
+whole makes backwards incompatible changes on a 2 version deprecation cycle.
+
+Several libraries have successfully migrated their user community to a new
+namespace with a version number on it, such as OpenCV (imported as ``cv2``) and
+BeautifulSoup (imported as ``bs4``), Jinja (``jinja2``) and psycopg (currently
+imported as ``psycopg2``). Further afield, R's ggplot is used as ``ggplot2``.
+
+Implementation
+--------------
+
+The details of the proposal are as follows:
+
+- scikit-image 0.19 will be followed by scikit-image 1.0. Every deprecation
+  message will be removed from 1.0, and the API will be considered the
+  scikit-image 1.0 API.
+- After 1.0, the main branch will be changed to (a) change the import name to
+  skimage2, (b) change the package name to skimage2, and (c) change the version
+  number to 2.0-dev.
```

Comment:
We discussed this in various zoom meetings, I'm not sure that it was ever written down.

I'm not very opposed to `scikit-image2` but with `skimage2` people were generally happy to finally have our package name match our import name. I think people would hesitate about whether to type `scikit-image2` or `scikit-image-2`, for one.

> What separates skimage2 from all other scikits?

Well, for one, if this SKIP goes through we will be the first scikit to ever have a number in its import name. So that's already a major point of difference and the "trying to harmonise all the scikits" ship has sailed. I don't think this is a huge goal, particularly after this SKIP.

Reply:
That is true. I think either name will have its drawbacks: for `skimage2` it is not as immediately obvious (to me) that its a successor. However, I can really see the practicality of a name that is consistent with the import name. I imagine that we don't want to change scikit-image.org or the org and repo names on GitHub? Also if we use `scikit-image2` we have another new name (other being `skimage2`) associated with the project. I'd like that name count to be as low as possible.

I think this and the reasoning behind this would be good points to explicitly discuss in the Alternatives section. I'm happy to do that if you want?

Reply:
Yeah, I would appreciate that, and then please add yourself to the authors list! :blush:

Reply:
(Yes, we intend to keep the website.)

Reply:
I think `skimage2` is a bit better than `scikit-image2` both for the matching package and import names as well as minimizing the number of new names being introduced. I would say `skimage2` is still "scikit-image", just v2 of the API. I think for current users who know what `skimage` is, just seeing the `skimage2` name in a tutorial or blog post code snippet somewhere, they would likely infer that it is a newer version of the package.

Reply:
I think @stefanv was advocating for `skimage2` as well.